forge/test/mlir/mnist/utils.py

```diff
@@ -14,32 +14,56 @@
 
 # Model definition
 class MNISTLinear(nn.Module):
-    def __init__(self, input_size=784, output_size=10, hidden_size=256):
+    def __init__(self, input_size=784, output_size=10, hidden_size=512, bias=True, dtype=torch.float32):
         super(MNISTLinear, self).__init__()
-        self.l1 = nn.Linear(input_size, hidden_size)
-        self.relu = nn.ReLU()
-        self.l2 = nn.Linear(hidden_size, output_size)
+        self.linear_relu_stack = nn.Sequential(
+            nn.Linear(input_size, hidden_size, bias=bias, dtype=dtype),
+            nn.ReLU(),
+            nn.Linear(hidden_size, hidden_size, bias=bias, dtype=dtype),
+            nn.ReLU(),
+            nn.Linear(hidden_size, output_size, bias=bias, dtype=dtype),
+        )
 
     def forward(self, x):
-        x = self.l1(x)
-        x = self.relu(x)
-        x = self.l2(x)
+        logits = self.linear_relu_stack(x)
+        return logits
 
-        return nn.functional.softmax(x)
 
+class EarlyStopping:
+    def __init__(self, patience=3):
+        self.patience = patience
+        self.counter = 0
+        self.best_score = None
+        self.early_stop = False
 
-def load_tb_writer():
+    def __call__(self, val_metric):
```

Comment:
Sometimes you want to save model with largest val metric (accuracy) and sometimes with lowest (loss). This need to be factored in.